tests/gcp/hooks/test_cloud_memorystore.py

```diff
@@ -0,0 +1,386 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Dict, Sequence, Tuple
+from unittest import TestCase, mock
+
+from google.api_core.retry import Retry
+from google.cloud.exceptions import NotFound
+from google.cloud.redis_v1.types import Instance
+
+from tests.contrib.utils.base_gcp_mock import (
+    GCP_PROJECT_ID_HOOK_UNIT_TEST,
+    mock_base_gcp_hook_default_project_id,
+    mock_base_gcp_hook_no_default_project_id,
+)
+
+from airflow import AirflowException, version
+from airflow.gcp.hooks.cloud_memorystore import CloudMemorystoreHook
+
+TEST_GCP_CONN_ID = "test-gcp-conn-id"  # type: str
+TEST_DELEGATE_TO = "test-delegate-to"  # type: str
+TEST_LOCATION = "test-location"  # type: str
+TEST_INSTANCE_ID = "test-instance-id"  # type: str
+TEST_INSTANCE = Instance(name="instance-name")  # type: Instance
+TEST_PROJECT_ID = "test-project-id"  # type:  str
+TEST_RETRY = Retry()  # type: Retry
+TEST_TIMEOUT = 10  # type: float
+TEST_METADATA = [("KEY", "VALUE")]  # type:  Sequence[Tuple[str, str]]
+TEST_PAGE_SIZE = 100  # type: int
+TEST_UPDATE_MASK = {"paths": ["memory_size_gb"]}  # type: Dict
+TEST_PARENT = "projects/test-project-id/locations/test-location"  # type: str
+TEST_NAME = "projects/test-project-id/locations/test-location/instances/test-instance-id"  # type: str
+TEST_PARENT_DEFAULT_PROJECT_ID = "projects/{}/locations/test-location".format(
+    GCP_PROJECT_ID_HOOK_UNIT_TEST
+)  # type: str
+TEST_NAME_DEFAULT_PROJECT_ID = "projects/{}/locations/test-location/instances/test-instance-id".format(
```

Comment:
Maybe it would be better to use path builder instead of string formatting?

Reply:
This is a test. I think constant value will be better because it will also allow us to find regression when the external library changes behavior. In a perfect world, we should mock the path generating functions in the hook.